src/meta/src/manager/catalog/database.rs

```diff
@@ -268,6 +268,14 @@ impl DatabaseManager {
         self.connections.get(&connection_id)
     }
 
+    // only used for [`meta::rpc::cloud_service`]
+    // use get_connection for efficiency
+    pub fn get_connection_by_name(&self, connection_name: &str) -> Option<&Connection> {
+        self.connections
```

Comment:
Would `self.connections` contain all connection object across different database or only one database? What if there are connections with same name in many databases?

Reply:
Changed to `connection.id` in the with properties argument